to.etc.domui/src/main/java/to/etc/domui/component/layout/TabPanelBase.java

```diff
@@ -282,8 +284,15 @@ private TabInstance addTabInstance(TabInstance ti, int pos) {
 			m_tablist.add(pos, ti);
 		}
 		m_tabBuilder = null;
-		if(isBuilt())
-			forceRebuild();
+		if(isBuilt()) {
+			renderLabel(pos, ti);
```

Comment:
renderLabel() seems to render the tab label always at the end, not at the "pos" position. So I think you need a fix there too.